SYCL/KernelFusion/abort_fusion.cpp

```diff
@@ -0,0 +1,114 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_RT_WARNING_LEVEL=1 %CPU_RUN_PLACEHOLDER %t.out 2>&1\
+// RUN: %CPU_CHECK_PLACEHOLDER
+// RUN: env SYCL_RT_WARNING_LEVEL=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1\
+// RUN: %GPU_CHECK_PLACEHOLDER
+// UNSUPPORTED: cuda || hip
+// REQUIRES: fusion
+
+// Test fusion being aborted: Different scenarios causing the JIT compiler
+// to abort fusion due to constraint violations for fusion. Also check that
+// warnings are printed when SYCL_RT_WARNING_LEVEL=1.
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+constexpr size_t dataSize = 512;
+
+enum class Internalization { None, Local, Private };
+
+template <typename Kernel1Name, typename Kernel2Name, int Kernel1Dim>
+void performFusion(queue &q, int *in1, int *in2, int *in3, int *tmp, int *out,
+                   range<Kernel1Dim> k1Global, range<Kernel1Dim> k1Local) {
+  {
+    buffer<int> bIn1{in1, range{dataSize}};
+    buffer<int> bIn2{in2, range{dataSize}};
+    buffer<int> bIn3{in3, range{dataSize}};
+    buffer<int> bTmp{tmp, range{dataSize}};
+    buffer<int> bOut{out, range{dataSize}};
+
```

Comment:
The reason for defining the content in `main` is to avoid repeated initialization of the input data (`in1` - `in3`), as this only needs to happen once.

Reply:
That's not a production implementation that customers see. I'd personally prefer readability in tests over "performance".

I'd like to see what others think about this though. 

+ @steffenlarsen

Reply:
As there are now less buffers to initialize, I moved the initialization to be close to the buffer definition.